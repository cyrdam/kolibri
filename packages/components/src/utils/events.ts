// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
