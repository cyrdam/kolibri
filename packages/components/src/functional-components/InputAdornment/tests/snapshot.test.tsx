import { h } from '@stencil/core';
import { renderFunctionalComponentToSpecPage } from '../../../utils/testing';
import KolInputAdornmentFc from '../InputAdornment';
import KolIconButtonFc from '../../IconButton';

describe('KolInputContainerFc', () => {
	it('should render correctly', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => <KolInputAdornmentFc />);
		expect(page.root).toMatchSnapshot();
	});

	it('should render with children', async () => {
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolInputAdornmentFc>
				<KolIconButtonFc componentName="icon" icon="start-icon" />
			</KolInputAdornmentFc>
		));
		expect(page.root).toMatchSnapshot();
	});
});
