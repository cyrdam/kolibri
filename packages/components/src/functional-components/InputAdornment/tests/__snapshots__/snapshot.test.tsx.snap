// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KolInputContainerFc should render correctly 1`] = `<div class="kol-input-container__adornment kol-input-container__adornment--start"></div>`;

exports[`KolInputContainerFc should render with children 1`] = `
<div class="kol-input-container__adornment kol-input-container__adornment--start">
  <kol-icon _icons="codicon codicon-start-icon" _label=""></kol-icon>
</div>
`;
